draw.aaline: blend correctly

dlon opened the pull request:
#575 

Difference:
![0](https://i.imgur.com/mZs5TIj.png)
![1](https://i.imgur.com/c7Flztp.png)

illume commented on src_c/draw.c:
```diff
@@ -102,10 +102,6 @@ aaline(PyObject *self, PyObject *arg)
         return NULL;
     surf = pgSurface_AsSurface(surfobj);
 
-    if (surf->format->BytesPerPixel != 3 && surf->format->BytesPerPixel != 4)
```
oh cool. I guess this means it supports more than 32/24 bit now.

dlon replied:
I think it's always between 1 and 4, but 2 and 1 also work now

e1000 commented on src_c/draw.c:
```diff
@@ -983,27 +975,75 @@ set_at(SDL_Surface *surf, int x, int y, Uint32 color)
+static void
+set_pixel_32(Uint8 *pixels, SDL_PixelFormat *format, Uint32 pixel)
+{
+    switch (format->BytesPerPixel) {
```
Cool! It looks similar to the code in **drawline** now, but why it is much more simple here? Can we use this code in *drawline* ?

dlon replied:
In drawline, the line is drawn for each case inside the switch statement, probably because it might be faster.

e1000 commented on src_c/draw.c:
```diff
@@ -983,27 +975,75 @@ set_at(SDL_Surface *surf, int x, int y, Uint32 color)
+                                 br * colorptr[1], \
+                                 br * colorptr[2], \
+                                 br * colorptr[3])); \
+    }}
```
maybe @illume would like to apply some *clang-format* ? ;)

dlon commented:
Some measurements:

Not inlined:
  18985 calls/s
Inlined:
  18854 calls/s
Replaced trunc
  23240 calls/s
Replaced trunc + inlined:
  22826 calls/s

Inlining seems to make little difference when I try it, but replacing trunc() made a big difference.

Also, the code for big endian might be wrong.

e1000 commented:
* `x` is for expected Failure,
* `u` is for unexpected Success,

so it seems, no unexpected success :-/ I will investigate that "ASAP"...
